### ISSUE TITLE
Don't delete activities when a card is removed

### ISSUE THREAD

**Issue by bseclier** [RestyaPlatform/board#1710]
Hi,
I don't know if it is a bug or if it's just not implemented but I found a problem :
When I delete a card, it would be nice to delete all activities related. I'm in 0.6.2 and it's not the case. Here is the proof :

```
restyaboard=# select id,type,card_id from activities where card_id=1601;
   id   |     type      | card_id 
--------+---------------+---------
 228012 | delete_card   |    1601
 228011 | archived_card |    1601
 228010 | add_comment   |    1601
 228009 | add_comment   |    1601
 228008 | add_comment   |    1601
 228007 | add_card      |    1601
```

Is it possible to correct/add this ?
Thanks,

**Comment by rrjanbiah** [comment]
@bseclier 

It's actually a feature. Without activities, you won't know what is happening in the board when there are many users.

**Comment by bseclier** [comment]
Yes, but what the point to keep all this stuff when a card is deleted ?

**Comment by rrjanbiah** [comment]
*Internal note*

Research more about this. Are we using soft delete? 

For me, it seems to work correct as we can check what is deleted from admin activities. But, admin can't seem to restore it

CC @ssivachidambaram

**Comment by bseclier** [comment]
Do you see any contraindication to delete those "orphan" activities with the following SQL request (fields maybe used somewhere else in restya) ?
`delete from activities where card_id not in (select id from cards);`

**Comment by rrjanbiah** [comment]
@bseclier I'm sorry. I think, you're right. Comments are actually getting hard-deleted. Intended behavior is that comments are getting soft-deleted so that admin can revert back when required.

**Comment by bseclier** [comment]
I don't understand your answer. You don't want to delete all activities related to a deleted card just to let the possibility to an admin to restore the card ? Can you tell me where an admin can restore a card, I don't see it.
Besides, I don't want (as admin) to be able to do such a restoration, can I do the DELETE sql statement above without damaging the database, somewhere else ?
Thanks for your answer.

**Comment by rrjanbiah** [comment]
@bseclier 

> You don't want to delete all activities related to a deleted card just to let the possibility to an admin to restore the card ?

Yes. In a typical enterprise setup, admin should be able to handle any vandalism

> Can you tell me where an admin can restore a card, I don't see it.

Yes, currently there is no such thing as it was mistakenly not done yet

> Besides, I don't want (as admin) to be able to do such a restoration, can I do the DELETE sql statement above without damaging the database, somewhere else ?

Perhaps have to setup a button for admin to prune such histories?

**Comment by bseclier** [comment]
Yes, this kind of button would be awesome !

**Comment by rrjanbiah** [comment]
Related #1712

**Comment by bseclier** [comment]
I used this command in order to purge my boards :
`delete from activities where card_id not in (select id from cards) AND card_id != 0;`

**Comment by srileka** [comment]
@bseclier Please take a backup of your Restyaboard before executing the below command to purge your boards

`select * from activities where card_id NOT IN (select id from cards) AND card_id != 0 OR card_id != NULL;`
